tests/net/icmpv4/src/main.c

```diff
@@ -272,6 +300,36 @@ void test_icmpv4(void)
 	net_icmpv4_unregister_handler(&echo_req_opt_handler);
 
 	net_pkt_unref(pkt);
+
+	/* ================ Echo Request with Bad Options  ================= */
+	net_icmpv4_register_handler(&echo_req_opt_handler);
+
+	pkt = net_pkt_alloc_with_buffer(NULL, sizeof(icmpv4_echo_req_opt_bad),
+					AF_UNSPEC, 0, K_SECONDS(1));
+	zassert_not_null(pkt, "Allocation failed");
+
+	net_pkt_set_ip_hdr_len(pkt, sizeof(struct net_ipv4_hdr));
+	net_pkt_write(pkt, icmpv4_echo_req_opt_bad,
+		      sizeof(icmpv4_echo_req_opt_bad));
+
+	net_pkt_cursor_init(pkt);
+
+	net_pkt_set_ipv4_opts_len(pkt, 4);
+	net_pkt_set_overwrite(pkt, true);
+
+	net_pkt_skip(pkt, sizeof(struct net_ipv4_hdr)); /* Header*/
+	net_pkt_skip(pkt, 4); /* Options length */
+
+	hdr = (struct net_ipv4_hdr *)pkt->buffer->data;
+
+	ret = net_icmpv4_input(pkt, hdr);
+
+	/** TESTPOINT: Check input */
+	zassert_true(!(ret != NET_DROP), "Packet should drop");
```

Comment:
Works but more readable would have been `zassert_true(ret == NET_DROP, "Packet should drop");` or `zassert_equal(ret, NET_DROP, "Packet should drop");`.
No need to change it just for this comment, but if there are other comments that require changes, please tweak this one a bit. Anyway, LGTM.

Reply:
done